### PR TITLE
Fix related media endpoint

### LINKED ISSUE #807: Related media view returns 500
## Description

It would appear that the related media page on `main` is returning 500s, here's the exception I get when visiting a `related_url` on `main`:

```python
web_1               | [2022-07-20 17:45:23,475 - django.request - 241][ERROR] [e95d22f7b9f1444e90f44b09f435c2be] Internal Server Error: /v1/images/c608767c-a696-4d07-9bc1-daf174ca724d/related/
web_1               | Traceback (most recent call last):
web_1               |   File "/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
web_1               |     response = get_response(request)
web_1               |   File "/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
web_1               |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
web_1               |   File "/venv/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
web_1               |     return view_func(*args, **kwargs)
web_1               |   File "/venv/lib/python3.10/site-packages/rest_framework/viewsets.py", line 125, in view
web_1               |     return self.dispatch(request, *args, **kwargs)
web_1               |   File "/venv/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
web_1               |     response = self.handle_exception(exc)
web_1               |   File "/venv/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
web_1               |     self.raise_uncaught_exception(exc)
web_1               |   File "/venv/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
web_1               |     raise exc
web_1               |   File "/venv/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
web_1               |     response = handler(request, *args, **kwargs)
web_1               |   File "/venv/lib/python3.10/site-packages/django/utils/decorators.py", line 46, in _wrapper
web_1               |     return bound_method(*args, **kwargs)
web_1               |   File "/api/catalog/api/views/media_views.py", line 104, in related
web_1               |     results, num_results = related_media(
web_1               |   File "/api/catalog/api/controllers/elasticsearch/related.py", line 57, in related_media
web_1               |     f"results={json.dumps(results)}"
web_1               |   File "/usr/local/lib/python3.10/json/__init__.py", line 231, in dumps
web_1               |     return _default_encoder.encode(obj)
web_1               |   File "/usr/local/lib/python3.10/json/encoder.py", line 199, in encode
web_1               |     chunks = self.iterencode(o, _one_shot=True)
web_1               |   File "/usr/local/lib/python3.10/json/encoder.py", line 257, in iterencode
web_1               |     return _iterencode(o, 0)
web_1               |   File "/usr/local/lib/python3.10/json/encoder.py", line 179, in default
web_1               |     raise TypeError(f'Object of type {o.__class__.__name__} '
web_1               | TypeError: Object of type Hit is not JSON serializable
```

On first glance it looks like this might have been introduced in #790

## Reproduction

1. `just up`
2. Visit http://localhost:8000/v1/images/?q=car
3. Click on the `related_url` for a result
4. See error.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #807  by @AetherUnbound 

## Description

Hit is not serializable, so we need to dump it to a dict like we do in the `search` module.

https://github.com/WordPress/openverse-api/blob/ce335722fe24ab574b463e227b965eab85621d43/api/catalog/api/controllers/elasticsearch/search.py#L248-L259

I also added integration tests for this. You can revert the change in `related.py` and see that the tests do fail so they are correctly testing the endpoint.

## Testing Instructions

Check that CI passes. Run `just api-test` locally. Try to reproduce the bug according to the linked issue and confirm you cannot.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — APPROVED** [review]
Works great, thank you for fixing this and for adding an integration test as well!

File: api/catalog/api/controllers/elasticsearch/related.py [hunk 1]
```diff
@@ -51,9 +52,16 @@ def related_media(uuid, index, filter_dead):
 
     result_count, _ = get_result_and_page_count(response, results, page_size)
 
+    dumpable_results = (
+        results.to_dict()
+        if isinstance(results, Hit)
+        else list(map(lambda r: r.to_dict(), results))
+    )
```
Comment (krysal) [inline api/catalog/api/controllers/elasticsearch/related.py:59]: nitpick: using a list comprehension would make it more readable.
```suggestion
    dumpable_results = (
        results.to_dict()
        if isinstance(results, Hit)
        else [r.to_dict() for r in results]
    )
```

Comment (sarayourfriend) [reply api/catalog/api/controllers/elasticsearch/related.py:59]: I.... have no idea why I reached for `map` there, lol! 100% JavaScript brain, I think. I'll update this in both places. Thanks, Krystle!

**Review by stacimc — APPROVED** [review]
I also prefer the list comprehension for readability, but verified both fix the bug.

**Pull request merged**